src/main/scala/mesosphere/marathon/api/CORSFilter.scala

```diff
@@ -0,0 +1,37 @@
+package mesosphere.marathon.api
+
+import javax.inject.{ Inject }
+import javax.servlet._
+import javax.servlet.http.{ HttpServletRequest, HttpServletResponse }
+import mesosphere.marathon.{ MarathonConf }
+
+import scala.collection.JavaConverters._
+
+class CORSFilter @Inject() (config: MarathonConf) extends Filter {
+  override def init(filterConfig: FilterConfig): Unit = {}
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+
+    response match {
+      case httpResponse: HttpServletResponse if config.accessControlAllowOrigin.isSupplied =>
+        val httpRequest = request.asInstanceOf[HttpServletRequest]
+        val static = ("Accept,Content-Type,Referer,Origin,Connection,Cache-Control,Access-Control-Request-Headers," +
+          "Pragma,Access-Control-Request-Method,Accept-Language,Accept-Encoding,User-Agent,Host,Accept-Language," +
+          "Accept-Encoding,Content-Length,Origin,X-DevTools-Emulate-Network-Conditions-Client-Id").split(",")
+        val staticSet = static.map(_.toLowerCase).toSet
```

Comment:
This value could be a lazy member of the companion object to avoid building a new Set for each response.